### PR TITLE
Development docs

### PR DESCRIPTION
**Pull request by lubomir**
Add general tips for creating new API. The goal is to have all the information summarized in one place, so that nothing is forgotten when new resources are added.

### CODE REVIEW

**Comment by bliuredhat** [comment]
+1

**Comment by xychu** [comment]
:+1:

File: docs/source/api_devel.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+This is a guide for adding new resources.
+
+
+Checklist
+---------
+
+#. Identify where to implement it: it can be part of existing application or
+   you can create a new application. If you want new application, use ::
+
```
Comment (ycheng-aa) [inline docs/source/api_devel.rst]: want new application -> want a new application?

**Comment by ycheng-aa** [comment]
Looks good.

**Comment by simozhan** [comment]
Looks good to me

File: docs/source/api_devel.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+   * ``$URL:release-list$`` → ``http://pdc.example.com/rest_api/v1/releases/``
+   * ``$URL:product-detail:dp$`` → ``http://pdc.example.com/rest_api/v1/products/dp/``
+
+ * ``$LINK:resourcename:param1:param2$`` will expand to clickable link to that
+   resource. The link label will be the URL of the resource (without the host
+   name).
+
+To describe available query filters, use ``%(FILTERS)s`` macro. This expands to
+an unordered list with filter names and types of the value. The serializer can
+be describe with ``%(SERIALIZER)s``, which expands to a code block with JSON
```
Comment (erichuanggit) [inline docs/source/api_devel.rst]: describe -> described

File: docs/source/api_devel.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+   same logic itself (report error when unknown field is specified).
+
+#. Write test cases for both success and error paths.
+
+
+Writing documentation
+---------------------
+
+The browsable documentation is exported from docstrings of view set methods. It
+uses `Markdown`_ as a markup language. There a couple helpers that make some
```
Comment (erichuanggit) [inline docs/source/api_devel.rst]: There -> There are

**Comment by erichuanggit** [comment]
Generally looks good to me.

**Pull request merged**